### PR TITLE
[FVM] Interaction limit error fix

### PR DESCRIPTION
**Pull request by janezpodhostnik**
Move of the internal PR: https://github.com/dapperlabs/flow-go/pull/6380

Should be merged as is, as this is whats currently deployed.

### CODE REVIEW

**Review by zhangchiqing — APPROVED** [review]

**Review by ramtinms — APPROVED** [review]

File: fvm/transactionInvoker.go [hunk 1]
```diff
@@ -75,7 +75,15 @@ func (i *TransactionInvoker) Process(
 			proc.Events = make([]flow.Event, 0)
 			proc.ServiceEvents = make([]flow.Event, 0)
 		}
-		if mergeError := parentState.MergeState(childState, sth.EnforceInteractionLimits()); mergeError != nil {
+
+		// backwards compatibility to enable rolling deploy
+		enforceLimits := sth.EnforceInteractionLimits()
+		if (ctx.Chain.ChainID() == flow.Mainnet && blockHeight > uint64(36419249)) ||
+			(ctx.Chain.ChainID() == flow.Testnet && blockHeight > uint64(78354421)) {
```
Comment (ramtinms) [inline fvm/transactionInvoker.go:82]: I wonder if we need these block heights?

Comment (janezpodhostnik) [reply fvm/transactionInvoker.go:82]: I would say yes if someone wants to rerun old blocks.

**Comment by janezpodhostnik** [comment]
bors merge

**Comment by janezpodhostnik** [comment]
bors retry

**Comment by janezpodhostnik** [comment]
bors retry

**Pull request merged**